src/project-manager.ts

```diff
@@ -731,33 +964,51 @@ export class ProjectConfiguration {
 		return null;
 	}
 
+	/**
+	 * @return language service object
+	 */
 	getService(): ts.LanguageService {
 		if (!this.service) {
 			throw new Error('project is uninitialized');
 		}
 		return this.service;
 	}
 
+	/**
+	 * Note that it does not perform any parsing or typechecking 
+	 * @return program object (cached result of parsing and typechecking done by TS service) 
+	 */
 	getProgram(): ts.Program {
 		if (!this.program) {
 			throw new Error('project is uninitialized');
 		}
 		return this.program;
 	}
 
+	/**
+	 * @return language service host that TS service uses to read the data 
+	 */
 	getHost(): InMemoryLanguageServiceHost {
 		if (!this.host) {
 			throw new Error('project is uninitialized');
 		}
 		return this.host;
 	}
 
+	/**
+	 * Tells TS service to recompile program (if needed) based on current list of files and compilation options.
+	 * TS service relies on information provided by language servide host to see if there were any changes in 
+	 * the whole project or in some files 
+	 */
 	syncProgram(): void {
 		this.program = this.getService().getProgram();
 	}
 
 	private initialized?: Promise<void>;
 
+	/**
+	 * Initializes (sub)project by parsing configuration and making proper internal objects
+	 */
 	private init(): Promise<void> {
```

Comment:
Is there a reason this is asynchronous? It looks like it doesn't do anything async, so couldn't this simply be the constructor of the object?